specification/cognitiveservices/Language.AnalyzeText/models/common.models.tsp

```diff
@@ -0,0 +1,987 @@
+  @maxValue(1)
+  @minValue(0)
+  @clientName("confidenceThreshold")
+  confidenceScoreThreshold?: float64;
+
+  @doc("Context object with previous QnA's information.")
+  @clientName("answerContext")
+  context?: KnowledgeBaseAnswerContext;
+
+  @doc("Type of ranker to be used.")
+  @clientName("rankerKind")
+  rankerType?: RankerKind;
+
+  @doc("Filter QnAs based on given metadata list and knowledge base sources.")
+  filters?: QueryFilters;
+
+  @doc("To configure Answer span prediction feature.")
+  @clientName("shortAnswerOptions")
+  answerSpanRequest?: ShortAnswerOptions;
+
+  @doc("(Optional) Flag to enable Query over Unstructured Sources.")
+  includeUnstructuredSources?: boolean;
+}
+
+@doc("Type of ranker to be used")
+union RankerKind {
+  string,
+
+  @doc("Default ranker.")
+  Default: "Default ranker",
+
+  @doc("Question only ranker.")
+  Question: "Question only ranker",
+}
+
+@doc("Context object with previous QnA's information.")
+model KnowledgeBaseAnswerContext {
+  @doc("Previous turn top answer result QnA ID.")
+  previousQnaId: int32;
+
+  @doc("Previous user query.")
+  @clientName("previousQuestion")
+  previousUserQuery?: string;
+}
+
+@doc("filters over knowledge base.")
+model QueryFilters {
+  @doc("filters over knowledge base.")
+  metadataFilter?: MetadataFilter;
+
+  @doc("filters over knowledge base.")
+  sourceFilter?: SourceFilter;
+
+  @doc("Logical operation used to join metadata filter with source filter.")
+  logicalOperation?: LogicalOperationKind = LogicalOperationKind.AND;
+}
+
+@doc("Find QnAs that are associated with the given list of metadata.")
+model MetadataFilter {
+  @doc("List of metadata.")
+  metadata?: Array<MetadataRecord>;
+
+  @doc("Operation used to join metadata filters.")
+  logicalOperation?: LogicalOperationKind = LogicalOperationKind.AND;
+}
+
+@doc("Object to provide the key value pair for each metadata.")
+model MetadataRecord {
+  @doc("Metadata Key from Metadata dictionary used in the QnA.")
+  key: string;
+
+  @doc("Metadata Value from Metadata dictionary used in the QnA.")
+  value: string;
+}
+
+@doc("Find QnAs that are associated with any of the given list of sources in knowledge base.")
+model SourceFilter is Array<string>;
+
+#suppress "@azure-tools/typespec-azure-core/no-closed-literal-union" // defined as closed enum
+@doc("Set to 'OR' or 'AND' for using corresponding logical operation.")
+union LogicalOperationKind {
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "The EnumMember named 'AND' should have a documentation or description, please use decorator @doc to add it."
+  AND: "AND",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "The EnumMember named 'OR' should have a documentation or description, please use decorator @doc to add it."
+  OR: "OR",
+}
+
+@doc("To configure Answer span prediction feature.")
+model ShortAnswerOptions {
+  @doc("Enable or disable Answer Span prediction.")
+  enable: true;
```

Comment:
```suggestion
  enable: boolean = true;
```

Reply:
We discussed this before and it was noted the this is the best way to describe this property because it is a constant value